src/Microsoft.Health.Dicom.Core/Features/Validation/DateTimeValidation.cs

```diff
@@ -0,0 +1,33 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using Dicom;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Extensions;
+
+namespace Microsoft.Health.Dicom.Core.Features.Validation
+{
+    internal class DateTimeValidation : ElementValidation
+    {
+        public override void Validate(DicomElement dicomElement)
+        {
+            base.Validate(dicomElement);
+
+            string value = dicomElement.Get<string>();
+            string name = dicomElement.Tag.GetFriendlyName();
+            if (string.IsNullOrEmpty(value))
+            {
+                return;
+            }
+
+            if (!DateTimeOffset.TryParseExact(value, Extensions.DicomDatasetExtensions.DateTimeFormatsDT, CultureInfo.InvariantCulture, DateTimeStyles.None, out _))
```

Comment:
Did you look at https://github.com/fo-dicom/fo-dicom/blob/development/DICOM/DicomValidation.cs?

Is this something we can use?

Reply:
thanks for referring to this. I will check it out and use this existing validation.

Reply:
Maybe we should utilize this validation method for all other VRs. I remember there was a bug where it fails to write (or maybe read) dicom data into json file, this change probably can solve that issue.